sspl_test/alerts/self_hw/test_node_ipmitool_func.py

```diff
@@ -0,0 +1,116 @@
+from alerts.self_hw.self_hw_utilities import get_manufacturer_name
+
+def init(args):
+    pass
+
+def test_ipmitool_version(args):
+    """Check for expected ipmitool & IPMI version and v2 compliant."""
+    expected_ipmitool_ver = "1.8.18"
+    expected_ipmi_version = "2.0"
+    expected_ipmi_compliant = "v2 compliant"
+    # Check ipmitool version
+    tool_ver_cmd = "ipmitool -V"    # ipmitool version 1.8.18
+    res_op, res_err, res_rc = SimpleProcess(tool_ver_cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        search_res = re.search(
+            r"ipmitool version[\s]+([\w.]+)(.*)", res_op)
+        if search_res:
+            version_found = search_res.groups()[0]
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+    if version_found != expected_ipmitool_ver:
+        print("Expected: %s Actual: %s" % (
+            expected_ipmitool_ver, version_found))
+        assert False
+
+    # Check IPMI version
+    ipmi_ver_cmd = "ipmitool bmc info"     # IPMI Version : 2.0
+    res_op, res_err, res_rc = SimpleProcess(ipmi_ver_cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        search_res = re.search(
+            r"IPMI Version[\s]+:[\s]+([\w.]+)(.*)", res_op)
+        if search_res:
+            version_found = search_res.groups()[0]
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+    if version_found != expected_ipmi_version:
+        print("Expected: %s Actual: %s" % (
+            expected_ipmi_version, version_found))
+        assert False
+
+    # Check IPMI v2 compliant
+    ipmi_compliant = "ipmitool sel info"   # Version : 1.5 (v1.5, v2 compliant)
+    res_op, res_err, res_rc = SimpleProcess(ipmi_compliant).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        search_res = re.search(
+            r"Version[\s]+:[\s]+.*(%s).*" % expected_ipmi_compliant, res_op)
+        if not search_res:
+            raise Exception(
+                "ERROR: IPMITOOL is not %s" % expected_ipmi_compliant)
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+
+def test_sensor_availability(args):
+    """Fail if any expected sensor is not detected by ipmitool."""
+    found_all_sensors = True
+    sensors = [
+        "Voltage",
+        "Temperature",
+        "Power Supply",
+        "Drive Slot / Bay",
+        "Fan"
+        ]
+    # Get manufacturer name
+    manufacturer = get_manufacturer_name()
+    for sensor in sensors:
+        cmd = ["ipmitool", "sdr", "type", sensor]
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc == 0:
+            res_op = res_op.decode().replace("\n", "")
+            if not res_op:
+                found_all_sensors = False
+                print(
+                    "'%s' sensor is not seen in %s node server." % (
+                        sensor, manufacturer))
+        else:
+            raise Exception("ERROR: %s" % res_err.decode())
```

Comment:
It will come out of the loop at first detection of missing sensor type itself. It should run for all required sensor types, so better split this test for each sensor, running through same code though reporting status per sensor type.

Reply:
No, it won't come out during 1st missing fru detection.
The raise exception is only if ipmitool rc !=0.
If sensor missing is detected still rc is 0 only.

> [root@smc1-m11 ~]# ipmitool sdr type 'Drive Slot / Bay'
> [root@smc1-m11 ~]#
> [root@smc1-m11 ~]# echo $?
> 0
>

Reply:
alright, then we are good